linkd-lib/src/cfg/seed.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright © 2021 The Radicle Link Contributors
+//
+// This file is part of radicle-link, distributed under the GPLv3 with Radicle
+// Linking Exception. For full terms see the included LICENSE file.
+
+use std::{convert::TryFrom, io, net::SocketAddr};
+
+use librad::net::discovery;
+use tokio::net::lookup_host;
+
+use librad::PeerId;
+
+use crate::args::Bootstrap;
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("the seed `{0}` failed to resolve to an address")]
+    DnsLookupFailed(String),
+
+    #[error(transparent)]
+    Io(#[from] io::Error),
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Seed {
+    pub addrs: Vec<SocketAddr>,
+    pub peer_id: PeerId,
+}
+
+impl Seed {
+    /// Create a [`Seed`] from a [`Bootstrap`].
+    ///
+    /// # Errors
+    ///
+    /// * If the supplied address cannot be resolved.
+    async fn try_from_bootstrap(bootstrap: &Bootstrap) -> Result<Self, Error> {
+        if let Some(addr) = lookup_host(bootstrap.addr.clone()).await?.next() {
```

Comment:
Why not `collect` so we can load multiple seeds here?

Reply:
Because they would share the same `peer_id` and that is a violation of the protocol. But you are absolutely right that we should collect all addresses and associate them with the same peer, as the discovery expects a stream of `(PeerId, Vec<SocketAddr>)`.

Reply:
9d993d9